TTbarAnalysis.py

```diff
@@ -713,117 +505,133 @@ def definePlots(self, t, noSel, sample=None, sampleCfg=None):
                 if make_TwoTagCountPlots:
-                    # to avoid refining the same selection in loops of ptBin
-                    TwoLeptonsTwoJets_METCut = TwoLeptonsTwoJets.refine("{0}Jets{1}selection_METcut".format(nbrj, channel), cut=[ corrMET.pt < 80.])
-                    plots +=(TwoTagCount(channel, jets, TwoLeptonsTwoJets_METCut, nbrj, era, isMC))
+                    for iPT in range(-1, len(lowerPtBinEdges)):
+                        ptBin = getPtLabel(iPT)
+                        if ptBin == "Inclusive":
+                            sel_twotag_pt = sel2l2j_METCut
+                        else:
+                            if ptBin =="600toInf":
+                                minpt = 600.
+                                maxpt = 1000.
+                            else:
+                                minpt = lowerPtBinEdges[iPT]
+                                maxpt = lowerPtBinEdges[iPT+1]
+                            sel_twotag_pt = sel2l2j_METCut.refine(f"{sel2l2j_METCut.name}_2Jets_{ptBin}ptBin_cut{nbrj}", cut=[ op.in_range(minpt, jets[0].pt, maxpt), op.in_range(minpt, jets[1].pt, maxpt) ])
+                        if isMC:
+                            nTrueB = op.rng_count(jets[:2], lambda j : j.hadronFlavour == 5)
+                        else:
+                            nTrueB = op.c_int(0)
+                        sel_twotag_pt_twoTrueB = sel_twotag_pt.refine(f"{sel_twotag_pt.name}_twoTrueB", cut=(nTrueB == 2))
+                        for tagger,workingPoints in bTaggingWPs_era[era].items():
+                            for wp,thresh in workingPoints.items():
+                                key = f"{tagger}{wp}"
+                                nTaggedB = op.rng_count(jets[:2], partial((lambda j,fun=None,val=None : fun(j) >= val), fun=t_bDisc[tagger], val=thresh))
+                                ## now two plots: two true b-jets -> n pass; two tagged: n true
+                                sel_twotag_pt_twoTaggedB = sel_twotag_pt.refine(f"{sel_twotag_pt.name}_twoTaggedB_{key}", cut=(op.rng_len(bjets_pass) == 2))
+                                plots += [
+                                    Plot.make1D(f"{channel}_Events_2l{nbrj}j_nbtagged_TruthFlav_2Beauty_{key}_{ptBin}",
+                                        nTaggedB, sel_twotag_pt_twoTrueB,
+                                        EqBin(3, 0., 3.), title="N b-tags for events with two true b's",
+                                        plotopts=utils.getOpts(channel, **{"log-y": True})),
+                                    Plot.make1D(f"{channel}_Events_2l{nbrj}j_2btagged_TruthFlav_nBeauty_{key}_{ptBin}",
+                                        nTrueB, sel_twotag_pt_twoTaggedB,
+                                        EqBin(3, 0., 3.), title="N true b's for events with two btags",
+                                        plotopts=utils.getOpts(channel, **{"log-y": True})),
```

Comment:
I am not sure about this histogram: I see 
`Events with 0 btagged +truth2b Flavour`  will be in bin 0 from the histogram just above 
`Events with 0 bFlav +2btagged ` will be in bin 0 as well 
 in addition, you have the data 

The same for the latter bins: The summed plot will bring us a lot of double counting 
I think we just need to go for more bins to add
 `0bFlav +2b-tagged `
`1bFlav + 2b-tagged` 
which represents for us the bin1 in the Old plot for events 2b-tagged with at least 1b truth flavour.
I am not sure if `op.rng_count` can work with `op.sum` that we can do something like
 `Plot.make1D(f"{channel}_Events_2l{nbrj}j_2btagged_TruthFlav_nBeauty_{key}_{ptBin}",
                                        op.sum(nTrueB, op.c_int(4)) if  nTrueB==op.c_int(0) else ( 
                                                  op.sum(nTrueB, op.c_int(5)) if  nTrueB==op.c_int(0) if                                                        nTrueB==op.c_int(0) ==1 else(
                                                         op.sum(nTrueB, op.c_int(5)) ))
                                        , sel_twotag_pt_twoTaggedB,
                                        EqBin(3, 0., 3.), title="N true b's for events with two btags",
                                        plotopts=utils.getOpts(channel, **{"log-y": True}))`
 and the data will be in op.c_int(-1) 
Then I am afraid that the data will be with same selection in the 2 histograms above: pass 2 lep +2btagged jets. so the summed histogram will be double count in bin -1, unless I am missing somthing about how the Summed Plots works.

Reply:
Okay, after looking more into this I see you're not adding the two histogram `pNTrueB_TwoTagged` and `pNTagged_TwoTrue` to one histogram so I think this should be fine for me !
thanks

Reply:
if you want them all in the same plot that should be possible (but then we need to change what ends up in which bin, and add them up as you did before).